### PR TITLE
MessageBox: add inputValue

### PR DESCRIPTION
**Pull request by Leopoldthecoder**
Please makes sure these boxes are checked before submitting your PR, thank you!

* [ ] Make sure you follow Element's contributing guide ([中文](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.zh-CN.md) | [English](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.en-US.md)).
* [ ] Make sure you are merging your commits to `dev` branch.
* [ ] Add some descriptions and refer relative issues for you PR.

#2748 #2865

### CODE REVIEW

File: packages/message-box/src/main.vue [hunk 1]
```diff
@@ -174,9 +174,14 @@
...
-        if (this.$type === 'prompt' && val !== null) {
-          this.validate();
+      inputValue: {
+        immediate: true,
+        handler(val) {
+          this.$nextTick(_ => {
+            if (this.$refs.input && this.$type === 'prompt' && val !== null) {
+              this.validate();
+            }
+          });
```
Comment (QingWei-Li) [inline packages/message-box/src/main.vue]: image

好蛋疼 immediate 后又 nextTick

Comment (Leopoldthecoder) [inline packages/message-box/src/main.vue]: 有啥好办法？

File: packages/message-box/src/main.vue [hunk 2]
```diff
@@ -174,9 +174,14 @@
...
 
     watch: {
-      inputValue(val) {
-        if (this.$type === 'prompt' && val !== null) {
-          this.validate();
+      inputValue: {
+        immediate: true,
+        handler(val) {
+          this.$nextTick(_ => {
+            if (this.$refs.input && this.$type === 'prompt' && val !== null) {
```
Comment (baiyaaaaa) [inline packages/message-box/src/main.vue]: inputValue 都改变了，为啥还要判断 this.$refs.input

Comment (Leopoldthecoder) [inline packages/message-box/src/main.vue]: inputValue 可能是用户传进来的初始值，需要立刻校验，不一定是通过输入框输入造成的改变。不过对于初始值，nextTick 里确实已经有 this.$refs.input 了，所以去掉了这个判断。

File: packages/message-box/src/main.vue (hunk 1 above)
Comment (QingWei-Li) [inline packages/message-box/src/main.vue]:
```js
created() {
  this.$watch('inputValue', val => this.$refs.input && this.$type === 'prompt' && val !== null && this.validate(), { immediate: true })
}
```

🌚

Comment (Leopoldthecoder) [inline packages/message-box/src/main.vue]: 达不到需要的效果吧，触发 `watch` 的时候 `this.$refs.input` 还是 `undefined`，于是就触发不了 `this.validate` 了。

**Pull request merged**